File: src/models/notebook-precautionary-measure.ts
import { Attachment } from "./attachment";

export type NotebookPrecautionaryMeasure = Readonly<{
  id: string;
  title: string;
  comments: string;
  reasons: string[];

  exclusion: boolean;

  circumstances: string;
  circumstancesDocuments: Attachment[];

  decisionMaker: string;
  giver: string;
  dateGiven: Date;
  startDate: Date;
  endDate: Date;

  // TODO: enum, see below
  disallowedAccesses: number[];
  // /** "à l'internat" */
  // Internat = 0,
  // /** "à l'établissement" */
  // Etablissement = 1,
  // /** "à la demi-pension" */
  // DP = 2

}>;

File: library/src/commonMain/kotlin/ink/literate/pawnote/models/NotebookPrecautionaryMeasure.kt
package ink.literate.pawnote.models

import kotlinx.datetime.LocalDateTime

data class NotebookPrecautionaryMeasure(
    val id: String,
    val title: String,
    val comments: String,
    val reasons: List<String>,

    val exclusion: Boolean,

    val circumstances: String,
    val circumstancesDocuments: List<Attachment>,

    val decisionMaker: String,
    val giver: String,
    val dateGiven: LocalDateTime,
    val startDate: LocalDateTime,
    val endDate: LocalDateTime,

    val disallowedAccesses: List<AccessKind>
)